storage/store_test.go

```diff
@@ -383,40 +362,73 @@ func TestStoreRangeIterator(t *testing.T) {
 		}
 	}
 
-	// Verify two passes of the iteration.
-	iter := newStoreRangeIterator(store)
+	// Verify two passes of the visit.
+	ranges := newStoreRangeSet(store)
 	for pass := 0; pass < 2; pass++ {
-		for i := 1; iter.EstimatedCount() > 0; i++ {
-			if rng := iter.Next(); rng == nil || rng.Desc().RaftID != int64(i) {
+		if ec := ranges.EstimatedCount(); ec != 10 {
+			t.Errorf("expected 10 remaining; got %d", ec)
+		}
+		i := 1
+		ranges.Visit(func(rng *Range) bool {
+			if rng.Desc().RaftID != int64(i) {
 				t.Errorf("expected range with Raft ID %d; got %v", i, rng)
 			}
+			if ec := ranges.EstimatedCount(); ec != 10-i {
+				t.Errorf("expected %d remaining; got %d", 10-i, ec)
+			}
+			i++
+			return true
+		})
+		if ec := ranges.EstimatedCount(); ec != 10 {
+			t.Errorf("expected 10 remaining; got %d", ec)
 		}
-		iter.Reset()
 	}
 
-	// Try iterating with an addition.
-	iter.Next()
-	if ec := iter.EstimatedCount(); ec != 9 {
+	// Try visiting with an addition and a removal.
+	visited := make(chan struct{})
+	updated := make(chan struct{})
+	done := make(chan struct{})
+	i := 1
```

Comment:
Ah, yes. Fixed.

Yes, this should be good now. Ran the tests again with `ScanInterval = 10 * time.Milisecond`, and all tests passed except `TestMultiStoreEventFeed`. Will merge after confirming all tests passed.